client/src/components/mapboxMap.tsx

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable no-unused-vars */
+import React, {useRef, useEffect, useState} from 'react';
+import {Map, NavigationControl} from 'mapbox-gl';
+import * as styles from './mapboxMap.module.scss';
+import mapStyle from '../data/mapStyle';
+import ZoomWarning from './zoomWarning';
+
+const MapboxMap = () => {
+  const mapContainer = React.useRef<HTMLDivElement>(null);
+  const map = useRef<Map>() as React.MutableRefObject<Map>;
+  const [lng, setLng] = useState(-86.502136);
+  const [lat, setLat] = useState(32.4687126);
+  const [zoom, setZoom] = useState(3);
+
+  useEffect(() => {
+    // Only initialize once
+    if (map.current && mapContainer.current) return;
```

Comment:
Gotta love stateful DOMs in javascript/ts

Reply:
yeah, this is a side effect of using `ref`s for sure - there might bet a better way to do this but I haven't come across this yet